File: Gasto/src/main/java/com/py/api/model/sistema/Licencia.java
package com.py.api.model.sistema;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Transient;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "SYS_LICENCIA")
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Licencia implements Serializable {

	private static final long serialVersionUID = 1374454165430519065L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ID_LICENCIA")
	private Long id;

	@Column(name = "SERIE")
	private String serie;

	@Column(name = "BO_ATIVO")
	private boolean ativo;

	@Column(name = "DT_ULTIMO_LOGIN")
	private Date fechaUltimoLogin;

	@Column(name = "DT_EXPIRACAO")
	private Date dtExpiracao;

	@Transient
	private String dtExpiracaoString;

	@Transient
	private Boolean programacao;

	@Transient
	private Long diasAVencer;

}

File: app_gasto_viejo/lib/src/module/login/models/licencia.dart
class Licencia {
  int? id;
  String? serie;
  bool? ativo;
  String? fechaUltimoLogin;
  String? dtExpiracao;
  String? dtExpiracaoString;
  bool? programacao;
  int? diasAVencer;

  Licencia({
    this.id,
    this.serie,
    this.ativo,
    this.fechaUltimoLogin,
    this.dtExpiracao,
    this.dtExpiracaoString,
    this.programacao,
    this.diasAVencer,
  });

  Licencia.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    serie = json['serie'];
    ativo = json['ativo'];
    fechaUltimoLogin = json['fechaUltimoLogin'];
    dtExpiracao = json['dtExpiracao'];
    dtExpiracaoString = json['dtExpiracaoString'];
    programacao = json['programacao'];
    diasAVencer = json['diasAVencer'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['serie'] = serie;
    data['ativo'] = ativo;
    data['fechaUltimoLogin'] = fechaUltimoLogin;
    data['dtExpiracao'] = dtExpiracao;
    data['dtExpiracaoString'] = dtExpiracaoString;
    data['programacao'] = programacao;
    data['diasAVencer'] = diasAVencer;
    return data;
  }

  Licencia copyWith({
    int? id,
    String? serie,
    bool? ativo,
    String? fechaUltimoLogin,
    String? dtExpiracao,
    String? dtExpiracaoString,
    bool? programacao,
    int? diasAVencer,
  }) {
    return Licencia(
      id: id ?? this.id,
      serie: serie ?? this.serie,
      ativo: ativo ?? this.ativo,
      fechaUltimoLogin: fechaUltimoLogin ?? this.fechaUltimoLogin,
      dtExpiracao: dtExpiracao ?? this.dtExpiracao,
      dtExpiracaoString: dtExpiracaoString ?? this.dtExpiracaoString,
      programacao: programacao ?? this.programacao,
      diasAVencer: diasAVencer ?? this.diasAVencer,
    );
  }
}
